src/test/java/teammates/test/cases/webapi/CreateFeedbackResponseActionTest.java

```diff
@@ -23,13 +50,192 @@ protected String getRequestMethod() {
+            a.execute();
+        });
+        ______TS("null responseDetail");
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            FeedbackResponseCreateRequest createRequest = getResponseRequest(student1InCourse1.getEmail());
+            createRequest.setResponseDetails(null);
+            Action a = getAction(createRequest, params);
+            a.execute();
+        });
+        ______TS("typical case for student");
+        loginAsStudent(student2InCourse1.getGoogleId());
+        String[] paramsQn2 = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn2InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+        };
+        FeedbackResponseCreateRequest createRequest = getResponseRequest(student1InCourse1.getEmail());
+        CreateFeedbackResponseAction typicalAction = getAction(createRequest, paramsQn2);
+        JsonResult typicalResult = getJsonResult(typicalAction);
+
+        assertEquals(HttpStatus.SC_OK, typicalResult.getStatusCode());
+        FeedbackResponseData typicalData = (FeedbackResponseData) typicalResult.getOutput();
+        assertEquals("This is the text", typicalData.getResponseDetails().getAnswerString());
+        assertEquals(student1InCourse1.getEmail(), typicalData.getRecipientIdentifier());
+        assertNotNull(typicalData.getFeedbackResponseId());
+        assertEquals(student2InCourse1.getEmail(), typicalData.getGiverIdentifier());
+
+        ______TS("typical case for instructor");
+        loginAsInstructor(instructor2OfCourse1.getGoogleId());
+        String[] paramsQn2Grace = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn2InGracePeriodInCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        FeedbackResponseCreateRequest requestForInstructor = getResponseRequest(instructor2OfCourse1.getEmail());
+        CreateFeedbackResponseAction typicalActionForInstructor = getAction(requestForInstructor, paramsQn2Grace);
+        JsonResult typicalResultInstructor = getJsonResult(typicalActionForInstructor);
+
+        assertEquals(HttpStatus.SC_OK, typicalResultInstructor.getStatusCode());
+        FeedbackResponseData typicalDataInstructor = (FeedbackResponseData) typicalResultInstructor.getOutput();
+        assertEquals("This is the text", typicalDataInstructor.getResponseDetails().getAnswerString());
+        assertNotNull(typicalDataInstructor.getFeedbackResponseId());
+        assertEquals(instructor2OfCourse1.getEmail(), typicalDataInstructor.getGiverIdentifier());
+        assertEquals(instructor2OfCourse1.getEmail(), typicalDataInstructor.getRecipientIdentifier());
+
+        ______TS("response already exists");
+        loginAsStudent(student2InCourse1.getGoogleId());
+        FeedbackResponseCreateRequest createRequestAlreadyExists = getResponseRequest(student5InCourse1.getEmail());
```

Comment:
Shouldn't `student5InCourse1.getEmail()` be `student2InCourse1.getGoogleId()`?

In addition, I think this test case has a pre-condition `The response already exists`. You need to either verify that or the test would have a coupling with the previous `______TS("typical case for student");`